src/guid.cpp

```diff
@@ -0,0 +1,33 @@
+// Luanti
+// SPDX-License-Identifier: LGPL-2.1-or-later
+// Copyright (C) 2024 SFENCE
+
+#include "guid.h"
+#include <sstream>
+
+#include "serverenvironment.h"
+#include "util/base64.h"
+
+GUIdGenerator::GUIdGenerator() :
+	m_uniform(0, UINT64_MAX)
+{
+	if (m_rand.entropy() <= 0.010)
+		throw BaseException("The system's provided random generator does not match "
+				"the entropy requirements for the GUId generator.");
+}
+
+GUId GUIdGenerator::next()
+{
+	std::stringstream s_guid;
+
+	s_guid << std::hex << std::setfill('0');
+	s_guid << std::setw(8) << (m_uniform(m_rand) & 0xFFFFFFFF) << "-";
```

Comment:
Side note (maybe premature optimization), many random bits are thrown away here. I would probably just generate two 64-bit random numbers and then extract bytes from them using bitops.